### PR TITLE
Add Habitica support for org mode

### PR DESCRIPTION
**Pull request by ksonney**
Cleaner merge request for adding habitica support to org-mode with habitrpg.el.

Key mappings work, the readme is fairly accurate, and you can get to the status screen with SPC-a-H to show your tasks and rewards and all that.

### CODE REVIEW

File: layers/+tools/habitrpg/packages.el [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+(defconst habitrpg-packages
+  '(
+    (habitrpg :location (recipe
+                         :fetcher github
+                         :repo "ryjm/habitrpg.el"))))
+
+(defun habitrpg/init-habitrpg ()
+  (require 'org-habit)
+  (require 'habitrpg))
```
Comment (TheBB) [inline layers/+tools/habitrpg/packages.el]: Please change this to use lazy loading.

File: layers/+tools/habitrpg/keybindings.el [hunk 2]
```diff
@@ -0,0 +1,3 @@
+;; Key Bindings for habitrpg.el
+(spacemacs/set-leader-keys "aH" 'habitrpg-status)
+(add-hook 'org-after-todo-state-change-hook 'habitrpg-add 'append)
```
Comment (TheBB) [inline layers/+tools/habitrpg/keybindings.el]: These should be in `packages.el` as part of the init function to which they belong.

File: layers/+tools/habitrpg/packages.el [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+;; - If PACKAGE is not referenced by any other Spacemacs layer, define a
+;;   function `habitrpg/init-PACKAGE' to load and initialize the package.
+
+;;; Code:
+
+(defconst habitrpg-packages
+  '(
+    (habitrpg :location (recipe
+                         :fetcher github
+                         :repo "ryjm/habitrpg.el"))))
```
Comment (TheBB) [inline layers/+tools/habitrpg/packages.el]: What's the problem with https://github.com/abrochard/emacs-habitica? That package is on MELPA.

Comment (ksonney) [inline layers/+tools/habitrpg/packages.el]: That particular package doesn't support org-mode, and this one does.

Comment (ksonney) [reply layers/+tools/habitrpg/packages.el]: @TheBB until I can convince the upstream maintainer to  submit the package to MELPA, is that an absolute blocker? As noted above, the linked package does not support org-mode, nor is it as feature complete.

**Pull request closed without merging**